### PR TITLE
Add consent selection validation and prompting for dashboard UI

### PR DESCRIPTION
**Pull request by thusithak**
Merge related with wso2/carbon-identity-framework/pull/1461

### CODE REVIEW

File: modules/jaggery-apps/portal/gadgets/consent_management/js/gadget.js [hunk 1]
```diff
@@ -312,23 +344,26 @@ function addActions(container) {
...
-            }
-        });
 
+        var selectedAttributes = container.jstree(true).get_selected();
+        var allSelected = compareArrays(allAttributes, selectedAttributes) ? true : false;
+
+        if(ALL_ATTRIBUTES_MANDATORY && isResidentIDP){
+            if (!allSelected) {
+                showWarning()
+            }else{
```
Comment (pulasthi7) [inline modules/jaggery-apps/portal/gadgets/consent_management/js/gadget.js]: Fix formatting

File: modules/jaggery-apps/portal/gadgets/consent_management/js/gadget.js [hunk 2]
```diff
@@ -350,6 +385,29 @@ function addActions(container) {
...
+        okCallback: function () {
+        }
+    });
+}
+
+function showConfirm(){
+    $("#message").html(confirmationDialog);
+    message({
+        title: "Consent Confirmation",
+        content: 'Are you sure you want to update/revoke this consent? This is not reversible.',
```
Comment (pulasthi7) [inline modules/jaggery-apps/portal/gadgets/consent_management/js/gadget.js]: shall we make this "....this operation is not reversible."

Comment (thusithak) [reply modules/jaggery-apps/portal/gadgets/consent_management/js/gadget.js]: Fixed with 9418a6ddb9355ce7854912891df37e75a94e5682

File: modules/jaggery-apps/portal/gadgets/consent_management/js/gadget.js (hunk 1 above)
Comment (thusithak) [reply modules/jaggery-apps/portal/gadgets/consent_management/js/gadget.js]: Fixed with 9418a6ddb9355ce7854912891df37e75a94e5682

**Pull request merged**